SonarQube.TeamBuild.PreProcessor/TeamBuildPreProcessor.cs

```diff
@@ -120,13 +124,12 @@ private void InstallLoaderTargets(ProcessedArgs args)
             }
         }
 
-        private bool FetchArgumentsAndRulesets(ProcessedArgs args, TeamBuildSettings settings, out IDictionary<string, string> serverSettings, out AnalyzerSettings analyzerSettings)
+        private bool FetchArgumentsAndRulesets(ISonarQubeServer server, ProcessedArgs args, TeamBuildSettings settings, out IDictionary<string, string> serverSettings, out AnalyzerSettings analyzerSettings)
         {
             string hostUrl = args.GetSetting(SonarProperties.HostUrl);
             serverSettings = null;
             analyzerSettings = null;
 
-            ISonarQubeServer server = this.factory.CreateSonarQubeServer(args, this.logger);
```

Comment:
Previously the server was being created here, but it's used by the build wrapper installer too so I moved the creation further up.